From 42cdb519686392c020a2eaefa0694c9df7569f88 Mon Sep 17 00:00:00 2001
From: Akira Ajisaka <aajisaka@apache.org>
Date: Wed, 30 Dec 2015 01:33:47 +0900
Subject: [PATCH 1057/2748] MAPREDUCE-6589. TestTaskLog outputs a log under
 directory other than target/test-dir. (aajisaka)

(cherry picked from commit 2a59d01503ed547bac15443fc273b5e04a2839d9)

Change-Id: I8cd6ffe2708dd3193e76d1390eb51b5a2bd84215
---
 .../java/org/apache/hadoop/mapred/TestTaskLog.java |   22 ++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapred/TestTaskLog.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapred/TestTaskLog.java
index ab4a342..ee3bd81 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapred/TestTaskLog.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapred/TestTaskLog.java
@@ -27,9 +27,10 @@
 import java.io.InputStream;
 
 import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.fs.FileUtil;
 import org.apache.hadoop.mapred.TaskLog.LogName;
-import org.apache.hadoop.mapreduce.MRJobConfig;
 import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.junit.AfterClass;
 import org.junit.Test;
 
 /**
@@ -37,6 +38,16 @@
  */
 public class TestTaskLog {
 
+  private static final String testDirName = TestTaskLog.class.getSimpleName();
+  private static final String testDir = System.getProperty("test.build.data",
+      "target" + File.separatorChar + "test-dir")
+      + File.separatorChar + testDirName;
+
+  @AfterClass
+  public static void cleanup() {
+    FileUtil.fullyDelete(new File(testDir));
+  }
+
   /**
    * test TaskAttemptID
    * 
@@ -67,7 +78,7 @@ public void testTaskLog() throws IOException {
     indexFile.delete();
     indexFile.createNewFile();
 
-    TaskLog.syncLogs("location", taid, true);
+    TaskLog.syncLogs(testDir, taid, true);
 
     assertTrue(indexFile.getAbsolutePath().endsWith(
         "userlogs" + File.separatorChar + "job_job_0001"
@@ -76,7 +87,7 @@ public void testTaskLog() throws IOException {
 
     f = TaskLog.getRealTaskLogFileLocation(taid, true, LogName.DEBUGOUT);
     if (f != null) {
-      assertTrue(f.getAbsolutePath().endsWith("location"
+      assertTrue(f.getAbsolutePath().endsWith(testDirName
           + File.separatorChar + "debugout"));
       FileUtils.copyFile(indexFile, f);
     }
@@ -84,14 +95,13 @@ public void testTaskLog() throws IOException {
     assertTrue(TaskLog.obtainLogDirOwner(taid).length() > 0);
     // test TaskLog.Reader
     assertTrue(readTaskLog(TaskLog.LogName.DEBUGOUT, taid, true).length() > 0);
-
   }
 
-  public String readTaskLog(TaskLog.LogName filter,
+  private String readTaskLog(TaskLog.LogName filter,
       org.apache.hadoop.mapred.TaskAttemptID taskId, boolean isCleanup)
       throws IOException {
     // string buffer to store task log
-    StringBuffer result = new StringBuffer();
+    StringBuilder result = new StringBuilder();
     int res;
 
     // reads the whole tasklog into inputstream
-- 
1.7.9.5

